### PR TITLE
Use BuckarooBanzay's luacheck dev branch

### PR DESCRIPTION
**Pull request by S-S-X**
Testing mt aware luacheck
* https://github.com/lunarmodules/luacheck/pull/108

### CODE REVIEW

**Comment by S-S-X** [comment]
Checking technic/machines/register/grinder_recipes.lua 1 warning
-    technic/machines/register/grinder_recipes.lua:143:24: accessing undefined field craftitem_place_item of global minetest

Checking technic/machines/register/recipes.lua    2 warnings
-    technic/machines/register/recipes.lua:259:14: accessing undefined field registered_on_mods_loaded of global minetest
-    technic/machines/register/recipes.lua:260:1: setting undefined field callback_origins.? of global minetest

**Comment by BuckarooBanzay** [comment]
> Checking technic/machines/register/grinder_recipes.lua 1 warning
> 
>     * technic/machines/register/grinder_recipes.lua:143:24: accessing undefined field craftitem_place_item of global minetest
> 
> 
> Checking technic/machines/register/recipes.lua 2 warnings
> 
>     * technic/machines/register/recipes.lua:259:14: accessing undefined field registered_on_mods_loaded of global minetest
> 
>     * technic/machines/register/recipes.lua:260:1: setting undefined field callback_origins.? of global minetest

IMO: this looks pretty good :thinking: the remaining fields should just be added to the local `.luacheckrc`

**Comment by alerque** [comment]
> IMO: this looks pretty good 🤔 the remaining fields should just be added to the local `.luacheckrc`

Are those undocumented/unreleased fields or 3rd party additions or something?

**Comment by S-S-X** [comment]
> Are those undocumented/unreleased fields or 3rd party additions or something?

TLDR: valid failures.

Well, I think `craftitem_place_item` been deprecated long ago, like maybe 10 years or something like that.

Direct use of `registered_on_mods_loaded` isn't exactly in best practices and this is one of those tables that is _kind of_ mentioned in docs but not exactly documented and you shouldn't mess with it... but well I guess dirty workaround to solve other issue was needed.

Direct use of `callback_origins` is a hack, shouldn't be done but ... well you know, has to do what it takes to make it work 😅
